### ISSUE TITLE
Upload/Download notification: show remaining time/number of files

### ISSUE THREAD

**Issue by tobiasKaminsky** [owncloud/android#1468]
Currently the notification of upload/download shows only progress.
I would like to enhance this by also showing:
- remaining time
- (x/Y) to indicate how many files are left to process. This will only be shown if there is more than one file in queue, resulting in:
120kB of 500kB, 5s remaining, (5/120)

@jancborchardt @AndyScherzinger what do the design gurus say? :)

**Comment by AndyScherzinger** [comment]
I tried to create a before/after mockup to get a basic idea.

today:
![today](https://cloud.githubusercontent.com/assets/1315170/12882163/0a4b24d0-ce50-11e5-9df6-348eac0e04d5.png)

adding multi-upload (+cancel) and remaining time
![new](https://cloud.githubusercontent.com/assets/1315170/12882175/1adb384e-ce50-11e5-8cd3-4826b8d1335e.png)

@jancborchardt is probably better at information design. :) I like the idea of having just one notification showing the overall progress when uploading multiple files. I wouldn't display time though since it is tricky to predict remaining time especially when bandwidth is drastically changing which might be the case for mobile. So I would go for remaining-files/total-files (x/y) and percentage. I also wouldn't display byte sizes since that is rather technical.

What do you think @jancborchardt ? - The cancel is just for a future thing to implement nad has nothing to do with this issue.

**Comment by jancborchardt** [comment]
Much better, yes! The current multiple notifications are quite annoying and not super useful. Wording seems fine, can’t really think of an improvement at the moment.

**Comment by enoch85** [comment]
Wow, much better!

**Comment by masensio** [comment]
Hi @all,

I like the idea of upload notification improvement, although I some doubts about that:
- Remainig time, it is a tricky data because it depends on the bandwidth (as @AndyScherzinger said)
- Name of current upload: I think it is useful to know what the current upload is
- Notifications in general: maybe after merging uploads view PR  the uploads notifications would be removed because we would have a complete view about the uploads status. In any case, we need to wait to decide it after merging uploads view PR.

**Comment by jancborchardt** [comment]
Regarding the name of current upload, we could have the title be:
> Uploading (2 of 4): IMG_22833.jpg

**Comment by AndyScherzinger** [comment]
True, issue to decide here would then be how to deal with long file names, mockups see below:

short name
![name1](https://cloud.githubusercontent.com/assets/1315170/13295201/0494cf02-db28-11e5-8f65-c1d013b3d6a6.png)

long name (middle ellipsized)
![name2](https://cloud.githubusercontent.com/assets/1315170/13295212/1068ccf2-db28-11e5-880c-ef7ed8c5c38e.png)

Thus the space is limited that's why I kept 2/4 for the counter for now since "2 of 4" depends on the language, German would be "von" and other might need even more characters thus eating up the space for the file name. Just a thought though.

**Comment by jancborchardt** [comment]
Lools good! Only thing I would change now is switch around to »5s remaining (79%)« so it's more of a sentence and a bit shorter.

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
